### ISSUE TITLE
Fail gracefully if SHELL is not defined.

### ISSUE THREAD

**Issue by jasonm23** [purcell/exec-path-from-shell#51]
Seems that some environments don't pass `SHELL`.

This causes the predicate which checks for standard shell to fail with a stringp / nil type error.

**Comment by jasonm23** [comment]
`exec-path-from-shell--standard-shell-p` the predicate in question.

**Comment by jasonm23** [comment]
I think it should throw an error like this:

```
(defun exec-path-from-shell--standard-shell-p (shell)
  "Return non-nil iff SHELL supports the standard ${VAR-default} syntax."
  (if (not (eq shell nil))
      (not (string-match "\\(fish\\|t?csh\\)$" shell))
    (error "SHELL is not defined")))
```

Or maybe just a warning:

```
(defun exec-path-from-shell--standard-shell-p (shell)
  "Return non-nil iff SHELL supports the standard ${VAR-default} syntax."
  (if (not (eq shell nil))
      (not (string-match "\\(fish\\|t?csh\\)$" shell))
    (message "Warning: SHELL is not defined")))
```

**Comment by jasonm23** [comment]
(of course the message option will need to return a nil too)

**Comment by jasonm23** [comment]
@purcell if you would prefer one way or the other, let me know.  I'm happy to code it if you don't have time.

**Comment by purcell** [comment]
Thanks. Addressed in 0f53502.

**Comment by jasonm23** [comment]
Thanks
